### PR TITLE
Require decoratorsBeforeExport option for decorators

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  | yes
| Minor: New Feature?      |
| Tests Added + Pass?      | 
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

It has not been decided yet where decorators should go.

### CODE REVIEW

File: packages/babel-parser/test/plugin-options.js [hunk 1]
```diff
@@ -14,9 +14,10 @@ describe("plugin options", function() {
     const OPT_2 = [NAME, { decoratorsBeforeExport: false }];
     const SYNTAX_1 = "@dec export class C {}";
     const SYNTAX_2 = "export @dec class C {}";
-    const SYNTAX_DEFAULT = "export @dec class C {}";
+    const SYNTAX_DEFAULT = "";
 
-    it("when they aren't specified", function() {
+    // Currently every option is required, so this can't be tested.
+    it.skip("when they aren't specified", function() {
```
Comment (xtuc) [inline packages/babel-parser/test/plugin-options.js]: Maybe we can just remove this dead code?

Comment (nicolo-ribaudo) [reply packages/babel-parser/test/plugin-options.js]: I forgot about flow's `all` option :laughing: I re-enabled the test using it.

**Comment by wyuenho** [comment]
This change makes no sense, `plugin-utils.js` throws as soon as `decoratorsBeforeExport` is undefined, but `babel-plugin-syntax-decorators/src/index.js` suggests that as long as `legacy: true` implies `decoratorsBeforeExport` is optional. Which exactly is the intention of this change?

**Comment by nicolo-ribaudo** [comment]
Are you using Babel or the parser? They have slightly different options:

| *Babel*                                                                    | *@babel/parser*                                       |
|----------------------------------------------------------------------------|-------------------------------------------------------|
| `["@babel/plugin-syntax-decorators", { "legacy": true }]`                  | `"decorators-legacy"`                                 |
| `["@babel/plugin-syntax-decorators", { "decoratorsBeforeExport": true }]`  | `["decorators", { "decoratorsBeforeExport": true }]`  |
| `["@babel/plugin-syntax-decorators", { "decoratorsBeforeExport": false }]` | `["decorators", { "decoratorsBeforeExport": false }]` |

**Pull request merged**